pyuvdata/uvflag.py

```diff
@@ -1798,49 +1970,96 @@ def to_baseline(self, uv, force_pol=False, run_check=True,
                 self.metric_array = self.metric_array.repeat(self.polarization_array.size, axis=-1)
-            self.weights_array = self.weights_array.repeat(self.polarization_array.size, axis=-1)
+                self.weights_array = self.weights_array.repeat(self.polarization_array.size, axis=-1)
         # Now the pol axes should match regardless of force_pol.
         if not np.array_equal(uv.polarization_array, self.polarization_array):
             if self.polarization_array.size == 1:
                 raise ValueError('Polarizations do not match. Try keyword force_pol'
                                  + ' if you wish to broadcast to all polarizations.')
             else:
                 raise ValueError('Polarizations could not be made to match.')
-        # Populate arrays
-        warr = np.zeros_like(uv.flag_array, dtype=np.float)
-        if self.mode == 'flag':
-            arr = np.zeros_like(uv.flag_array)
-            sarr = self.flag_array
-        elif self.mode == 'metric':
-            arr = np.zeros_like(uv.flag_array, dtype=float)
-            sarr = self.metric_array
-        for i, t in enumerate(np.unique(uv.time_array)):
-            ti = np.where(uv.time_array == t)
-            arr[ti, :, :, :] = sarr[i, :, :][np.newaxis, np.newaxis, :, :]
-            warr[ti, :, :, :] = self.weights_array[i, :, :][np.newaxis, np.newaxis, :, :]
-        if self.mode == 'flag':
-            self.flag_array = arr
-        elif self.mode == 'metric':
-            self.metric_array = arr
-        self.weights_array = warr
-
-        self.baseline_array = uv.baseline_array
-        self.Nbls = np.unique(self.baseline_array).size
+        if self.type == "waterfall":
+            # Populate arrays
+            if self.mode == 'flag':
+                arr = np.zeros_like(uv.flag_array)
+                sarr = self.flag_array
+            elif self.mode == 'metric':
+                arr = np.zeros_like(uv.flag_array, dtype=float)
+                warr = np.zeros_like(uv.flag_array, dtype=np.float)
+                sarr = self.metric_array
+            for i, t in enumerate(np.unique(self.time_array)):
+                ti = np.where(uv.time_array == t)
+                arr[ti, :, :, :] = sarr[i, :, :][np.newaxis, np.newaxis, :, :]
+                if self.mode == "metric":
+                    warr[ti, :, :, :] = self.weights_array[i, :, :][np.newaxis, np.newaxis, :, :]
+            if self.mode == 'flag':
+                self.flag_array = arr
+            elif self.mode == 'metric':
+                self.metric_array = arr
+                self.weights_array = warr
+
+        elif self.type == "antenna":
+            if self.mode == "metric":
+                raise NotImplementedError("Cannot currently convert from "
+                                          "antenna type, metric mode to "
+                                          "baseline type UVFlag object.")
+            else:
+                if uv.Nants_data > self.Nants_data:
```

Comment:
Is it reasonable to make this a little smarter? In principle you could have `uv.Nants_data < self.Nants_data`, but not all antennas in `uv` are in `self`.
In fact, I think you just need to run the `unique` and `setdiff1d` statements below and if `new_ants` has more than zero antennas, you do the rest.

Reply:
yeah that makes a lot more sense.